api/v1/views/instance.py

```diff
@@ -99,6 +99,9 @@ def get(self, request, provider_uuid, identity_uuid):
         if not esh_driver:
             return invalid_creds(provider_uuid, identity_uuid)
         identity = Identity.objects.get(uuid=identity_uuid)
+        # Probably redundant
+        if not user.can_use_identity(identity.id):
+            return invalid_creds(provider_uuid, identity_uuid)
```

Comment:
Does `not esh_driver` catch this case as well? I'm curious why it might be redundant.